### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by tpazderka**
- [ ] Any changes relevant to users are recorded in the `CHANGELOG.md`.
- [ ] The documentation has been updated, if necessary.
---

### CODE REVIEW

**Comment by tpazderka** [comment]
i am going to ignore the rest of the Codacy issues as they do not make much sense. We are not worried about pinning the dependencies and we have to do two runs of apt-get install so there is no sense in clearing the cache in-between the runs.

File: docker/op/cert.pem [hunk 1]
```diff
@@ -0,0 +1,28 @@
+-----BEGIN CERTIFICATE-----
```
Comment (schlenk) [inline docker/op/cert.pem]: I usually prefer to just create a self signed cert on the fly. This one is only valid till Oct 2019, so would this work after that date too?

Comment (tpazderka) [reply docker/op/cert.pem]: OK, I will make a separate issue for that.

Comment (schlenk) [reply docker/op/cert.pem]: yes, sounds fine. For testing its not really an issue.

**Pull request merged**